core/src/test/scala/unit/kafka/server/FetchRequestTest.scala

```diff
@@ -169,6 +171,58 @@ class FetchRequestTest extends BaseRequestTest {
   }
 
   /**
+   * Tests that down-conversions dont leak memory. Large down conversions are triggered
+   * in the server where the client closes connection. If references to byte arrays are
+   * not released this will OOM.
+   */
+  @Test
+  def testDownConversionWithConnectionFailure(): Unit = {
+    val (topicPartition, leaderId) = createTopics(numTopics = 1, numPartitions = 1).head
+
+    val producer = TestUtils.createNewProducer(TestUtils.getBrokerListStrFromServers(servers),
+      retries = 5, keySerializer = new StringSerializer, valueSerializer = new ByteArraySerializer)
```

Comment:
Do we want to tweak the producer batch.size and linger.ms?

Reply:
Since the messages are large, I thought the defaults are ok.

Reply:
The only thing is that it means that every batch will have a single message. It's probably fine for this test, but if we ever have a leak that only manifests itself if a batch has multiple messages, we won't catch it. So maybe we should also include some batches with multiple messages since it's easy enough to do? What do you think?

Reply:
Done.